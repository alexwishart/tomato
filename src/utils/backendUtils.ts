export const submitContactDetails = (name: string, email: string) => {
  //TODO
};
