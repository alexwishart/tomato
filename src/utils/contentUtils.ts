export const processTomatoCitations = (input: string, citationRoot: string) => {
  const citations = input.match(/\[[0-9]*\]/g);
  let output = input;

  // Assumes that citation numbers are unique in an input.
  if (citations) {
    citations.forEach((citation) => {
      const citationNumber = citation.substring(1, citation.length - 1);
      output = output.replace(
        citation,
        `<a href=${citationRoot}${citationNumber}>${citation}</a>`
      );
    });
    return output;
  } else {
    return input;
  }
};
