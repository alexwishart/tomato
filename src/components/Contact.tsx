import React, { useState } from 'react';
import styled from 'styled-components';
import { addSyntheticLeadingComment } from 'typescript';
import { contactContent } from '../content/contact';
import { submitContactDetails } from '../utils/backendUtils';

const Contact = () => {
  const [contactDetails, setContactDetails] = useState(['', '']);

  return (
    <Container>
      <h2>{contactContent.title}</h2>
      <p>{contactContent.subtitle}</p>
      <Form
        onSubmit={(e) => {
          e.preventDefault();
          submitContactDetails(contactDetails[0], contactDetails[1]);
        }}
      >
        {contactContent.inputFields.map((field, i) => {
          return (
            <Label key={field.value}>
              {field.label}
              <Input
                type="text"
                value={contactDetails[i]}
                onChange={(e) => {
                  const newDetails = [...contactDetails];
                  newDetails[i] = e.target.value;
                  setContactDetails(newDetails);
                }}
              ></Input>
            </Label>
          );
        })}
        <SubmitButton type="submit">
          {contactContent.submitButtonLabel}
        </SubmitButton>
      </Form>
    </Container>
  );
};

const Container = styled.div`
  width: 30%;
  padding-top: 1rem;
`;

const Form = styled.form`
  display: flex;
  flex-direction: column;
  width: 50%;
`;

const Label = styled.label`
  display: flex;
  align-items: center;
`;

const Input = styled.input`
  flex: 1;
  margin: 0.5rem;
`;

const SubmitButton = styled.button`
  padding: 0;
  font: inherit;
  cursor: pointer;
  border: none;
  outline: none;
  background: none;
  text-decoration: underline;

  align-self: flex-end;
  margin-top: 0.5rem;
`;

export default Contact;
