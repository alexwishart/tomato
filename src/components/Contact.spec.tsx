import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import Contact from './Contact';
import { act } from 'react-dom/test-utils';
import { submitContactDetails } from '../utils/backendUtils';

jest.mock('../utils/backendUtils');

afterEach(() => {
  jest.clearAllMocks();
});

describe('The Contact component', () => {
  it('should render correctly', () => {
    const { container } = render(<Contact />);
    expect(container).toMatchSnapshot();
  });

  it('should call the backend with the correct values when submit is clicked', () => {
    const { getByLabelText, getByText } = render(<Contact />);
    const nameInput = getByLabelText('Name');
    const emailInput = getByLabelText('E-mail');
    const submitButton = getByText('Submit');

    act(() => {
      fireEvent.change(nameInput, { target: { value: 'name' } });
    });

    act(() => {
      fireEvent.change(emailInput, { target: { value: 'name@test.com' } });
    });

    fireEvent.click(submitButton);
    expect(submitContactDetails).toHaveBeenCalledWith('name', 'name@test.com');
  });
});
