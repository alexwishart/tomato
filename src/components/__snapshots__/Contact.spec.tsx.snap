// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Contact component should render correctly 1`] = `
.c0 {
  width: 30%;
  padding-top: 1rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 50%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin: 0.5rem;
}

.c4 {
  padding: 0;
  font: inherit;
  cursor: pointer;
  border: none;
  outline: none;
  background: none;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  margin-top: 0.5rem;
}

<div>
  <div
    class="c0"
  >
    <h2>
      Keep up to date on the tomato
    </h2>
    <p>
      Enter your name and email address below to receive weekly updates on the status of the tomato
    </p>
    <form
      class="c1"
    >
      <label
        class="c2"
      >
        Name
        <input
          class="c3"
          type="text"
          value=""
        />
      </label>
      <label
        class="c2"
      >
        E-mail
        <input
          class="c3"
          type="text"
          value=""
        />
      </label>
      <button
        class="c4"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
